.dotnet.azure/README.md

```diff
@@ -494,6 +494,55 @@ _ = await assistantClient.DeleteAssistantAsync(assistant);
 _ = await assistantClient.DeleteThreadAsync(thread);
 ```
 
+## Advanced Topics
+
+### Using non-default service API versions
+
+Service API versions, as provided over REST via the `api-version` query string parameter, control what features are
+available and how those features behave. `AzureOpenAIClient` will automatically use the latest service API version
+that the current version of the library was built and validated against; it is strongly recommend to retain this
+default if using a different version is not otherwise necessary.
+
+> [!WARNING]
+> Using a non-default service API version may result in errors, incomplete or incorrect responses, or other unexpected
+> behavior in cases where capabilities or operation payloads changed between versions. Exercise caution if providing
+> a non-default value and consider using protocol method variants (see below) for maximum compatibility.
+
+In situations where using a non-default service API version is expressly desired, the default value may be changed by
+setting the `Version` property on `AzureOpenAIClientOptions` before a client is instantiated on the instance. As an
+example, the below will force the `2024-07-01-preview` service API label to be used:
+
+```csharp Snippet:ConfigureClient:NonDefaultVersion
+AzureOpenAIClientOptions optionsWithVersion = new()
+{
+    Version = "2024-07-01-preview",
```

Comment:
We do!

I can add an example of that somewhere, too, but e.g. for this we could use `ServiceVersion.V20240701Preview`. I just demonstrated with the wrong string because I think that might be a common reason to do it (custom versions that aren't (at least yet) enumerated).